packages/next-app/src/app/stableswapsimulator/(components)/ImpactCurve.tsx

```diff
@@ -0,0 +1,233 @@
+"use client";
+
+import { StableMath } from "@balancer-pool-metadata/math/src";
+
+import Plot from "#/components/Plot";
+import { Spinner } from "#/components/Spinner";
+import { useStableSwap } from "#/contexts/StableSwapContext";
+
+export function ImpactCurve() {
+  const {
+    baselineData,
+    variantData,
+    indexAnalysisToken,
+    indexCurrentTabToken,
+    preparePoolPairData,
+    numberToOldBigNumber,
+  } = useStableSwap();
+
+  if (
+    !baselineData ||
+    !baselineData.swapFee ||
+    !baselineData.ampFactor ||
+    !baselineData.tokens
+  )
+    return <Spinner />;
+
+  const tokensSymbol = baselineData.tokens.map((token) => token.symbol);
+
+  const calculateTokenAmounts = ({
```

Comment:
I think is just a preference, I'm comfortable doing the way you proposed, but for `calculateTokenAmounts` we need to wait for the math PR because we are using `preparePoolPairData` from the context at the moment xD

Adding a todo comment there